be/src/exec/pipeline/sink/sink_io_buffer.h

```diff
@@ -79,14 +79,9 @@ class SinkIOBuffer {
         return Status::OK();
     }
 
-    virtual bool is_finished() { return _is_finished; }
+    virtual bool is_finished() { return _is_finished && _num_pending_chunks == 0; }
```

Comment:
_num_pending_chunks == 0 can be reached on every error or normal cases?

Reply:
> _num_pending_chunks == 0 can be reached on every error or normal cases?

yes, in both normal and abnormal cases, all chunks must be processed once. The difference between them is that when an error occurs, cancel will be triggered. at this time, `_process_chunk` will only update `_num_pending_chunks` and will not really process chunk.